Services/UI/OJS.Services.Ui.Business/Implementations/ContestsBusinessService.cs

```diff
@@ -50,6 +50,60 @@ public class ContestsBusinessService : IContestsBusinessService
             this.contestCategoriesCache = contestCategoriesCache;
         }
 
+        public async Task<RegisterUserForContestServiceModel> RegisterUserForContest(int id, bool official)
+        {
+            var user = this.userProviderService.GetCurrentUser();
+            var userProfile = await this.usersBusinessService.GetUserProfileById(user.Id);
+
+            var participant = await this.participantsData
+                .GetWithContestByContestByUserAndIsOfficial(
+                    id,
+                    userProfile.Id,
+                    official);
+
+            var contest = this.contestsData
+                .GetByIdQuery(id)
```

Comment:
It is time to start moving out of `IQueryable<T>` results from data services.
For now, just create a method `GetById()` that returns the element itself.

Also, restrain from using methods that return `IQueryable`